### ISSUE TITLE
Space-insensitive search

### ISSUE THREAD

**Issue by neongreen** [aelve/codesearch#36]
Space between tokens usually doesn't matter, so I often end up having to write stuff like `import +Foo +(...` instead of `import Foo (...`. It would be nice to have a checkbox (might even be on by default) which would replace spaces with `<space>*` and which would allow any amount of spaces before/after punctuation.

**Comment by neongreen** [comment]
Some test-cases:

* `foo bar qux` should become `foo +bar +qux` (spaces become one-or-more spaces)
* `foo  bar qux` should become `foo  bar +qux` (when there is more than one space, we assume the author intended to avoid space translation)
* `foo\ bar` should become `foo\ bar` (escaping a space is another way to avoid space translation)
* `foo\\ bar` should become `foo\\ +bar` (in this case the space is not escaped but the backslash is)

Before starting to implement this task, I would recommend submitting another PR that would rewrite regex-handling logic, without implementing space-handling logic specifically. The reason for this recommendation is that it's easier to perform such operations on regexes if they are tokenized first. For instance, you could represent regexes as a sequence of tokens (in pseudocode):

```
(foo bar|\)qux)

Control("("), Literal "foo bar", Control("|"), Punctuation(')'), Literal("qux"), Control(")")
```

Then you can parse regexes to such token sequences, and render those token sequences back. After that all operations that we wish to do on regexes become trivial.

* To do case-insensitive search, just lowercase all `Literal`s and add a flag in the end.
* To do space-insensitive search, just look for all single spaces in `Literal`s and replace them with `[Literal(" "), Control("+")]`.
* To do precise search, just convert controls and punctuation to literals.

**Comment by neongreen** [comment]
And, of course, a small testsuite for this lexer would be a cool thing to have.
